src/Server/ArrowFlightHandler.cpp

```diff
@@ -280,327 +900,469 @@ UInt16 ArrowFlightHandler::portNumber() const
     return address_to_listen.port();
 }
 
-arrow::Status ArrowFlightHandler::ListFlights(
-    const arrow::flight::ServerCallContext &, const arrow::flight::Criteria *, std::unique_ptr<arrow::flight::FlightListing> * listings)
-{
-    std::vector<arrow::flight::FlightInfo> flights;
-    *listings = std::make_unique<arrow::flight::SimpleFlightListing>(std::move(flights));
-    return arrow::Status::OK();
-}
-
 arrow::Status ArrowFlightHandler::GetFlightInfo(
-    const arrow::flight::ServerCallContext & /*context*/,
-    const arrow::flight::FlightDescriptor & /*request*/,
-    std::unique_ptr<arrow::flight::FlightInfo> * /*info*/)
-{
-    return arrow::Status::OK();
-}
-
-arrow::Status ArrowFlightHandler::PollFlightInfo(
-    const arrow::flight::ServerCallContext & arrow_context,
+    const arrow::flight::ServerCallContext & context,
     const arrow::flight::FlightDescriptor & request,
-    std::unique_ptr<arrow::flight::PollInfo> * info)
+    std::unique_ptr<arrow::flight::FlightInfo> * info)
 {
-    (void)arrow_context;
-    (void)request;
-    (void)info;
-    return arrow::Status::OK();
+    auto impl = [&]
+    {
+        LOG_INFO(log, "GetFlightInfo is called for descriptor {}", request.ToString());
+
+        std::vector<arrow::flight::FlightEndpoint> endpoints;
+        int64_t total_rows = 0;
+        int64_t total_bytes = 0;
+        std::shared_ptr<const CHColumnToArrowColumn> ch_to_arrow_converter;
+
+        if ((request.type == arrow::flight::FlightDescriptor::CMD) && hasPollDescriptorPrefix(request.cmd))
```

Comment:
Does it make sense to accept poll descriptors in GetFlightInfo? My concern is the evaluation is not completely done, then:

It won't return all the tickets for the whole dataset
It is impossible to get the next poll descriptor
So I think this should result in an error here.